spec/hashie/dash_spec.rb

```diff
@@ -436,6 +436,25 @@ class KeepingMash < Hashie::Mash
       end
     end
   end
+
+  describe '#to_h' do
+    it 'writes out keys with nil values' do
+      expect(DashTest.new(first_name: 'Bob', email: nil).to_h).to eq(
+        first_name: 'Bob',
+        email: nil,
+        count: 0
+      )
+    end
+  end
+
+  describe '#to_hash' do
```

Comment:
I don't think this test is right... Why should `to_hash` work differently to `to_h`? It doesn't for normal hashes...

```
irb(main):002:0> { foo: "bar", baz: nil }.to_h
=> {:foo=>"bar", :baz=>nil}
irb(main):003:0> { foo: "bar", baz: nil }.to_hash
=> {:foo=>"bar", :baz=>nil}
```

I think it was mentioned that internal Hashie code treats them differently, but to let that implementation leak out into the contract is a shame.